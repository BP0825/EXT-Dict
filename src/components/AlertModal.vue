<template>
  <transition name="fade">
    <div class="modal show text-left" v-if="title || content">
      <div class="modal-dialog">
        <div class="modal-content">
          <div class="modal-header" v-if="title">
            <button type="button" class="close" @click="handleCancel">&times;</button>
            <h4 class="modal-title">{{ title }}</h4>
          </div>
          <div class="modal-body" v-if="content">
            <p>{{ content }}</p>
          </div>
          <div class="modal-footer">
            <button type="button" class="btn btn-default" @click="handleCancel">{{ i18n('cancel') }}</button>
            <button type="button" class="btn btn-danger" @click="handleConfirm">{{ i18n('confirm') }}</button>
          </div>
        </div>
      </div>
    </div>
  </transition>
</template>

<script>
export default {
  data () {
    return {
      title: '',
      content: '',
      onConfirm: null,
      onCancel: null
    }
  },
  methods: {
    i18n (key) {
      return chrome.i18n.getMessage(key)
    },
    handleConfirm () {
      if (typeof this.onConfirm === 'function') {
        this.onConfirm()
        this.onConfirm = null
      }
      this.title = ''
      this.content = ''
    },
    handleCancel () {
      if (typeof this.onCancel === 'function') {
        this.onCancel()
        this.onCancel = null
      }
      this.title = ''
      this.content = ''
    }
  },
  created () {
    this.$on('show', ({title, content, onConfirm, onCancel}) => {
      this.title = title
      this.content = content
      this.onConfirm = onConfirm
      this.onCancel = onCancel
    })
  }
}
</script>


<style lang="scss" scoped>
.fade-enter-active, .fade-leave-active {
  transition: opacity 500ms;
}
.fade-enter, .fade-leave-active {
  opacity: 0;
}
</style>

