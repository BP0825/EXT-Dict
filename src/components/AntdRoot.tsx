import React, { FC, useEffect } from 'react'
import { Helmet } from 'react-helmet'
import { Provider as ReduxProvider } from 'react-redux'
import createStore from '@/content/redux/create'
import { useRefFn, useObservableState, useObservable } from 'observable-hooks'
import { distinctUntilChanged, map, startWith } from 'rxjs/operators'

import SaladBowlContainer from '@/content/components/SaladBowl/SaladBowl.container'
import DictPanelContainer from '@/content/components/DictPanel/DictPanel.container'
import WordEditorContainer from '@/content/components/WordEditor/WordEditor.container'

import { createConfigStream } from '@/_helpers/config-manager'
import { reportGA } from '@/_helpers/analytics'
import { I18nContextProvider } from '@/_helpers/i18n'

import { ConfigProvider as AntdConfigProvider } from 'antd'
import zh_CN from 'antd/lib/locale-provider/zh_CN'
import zh_TW from 'antd/lib/locale-provider/zh_TW'
import en_US from 'antd/lib/locale-provider/en_US'

import 'antd/dist/antd.css'

const antdLocales = {
  'zh-CN': zh_CN,
  'zh-TW': zh_TW,
  en: en_US
}

const darkTheme =
  process.env.NODE_ENV === 'development'
    ? `https://cdnjs.cloudflare.com/ajax/libs/antd/4.1.0/antd.dark.min.css`
    : `/assets/antd.dark.min.css`

export interface AntdRootProps {
  /** analytics path */
  path?: string
}

export const AntdRoot: FC<AntdRootProps> = props => {
  const storeRef = useRefFn(createStore)

  const { locale, darkMode, analytics } = useObservableState(
    useObservable(() =>
      createConfigStream().pipe(
        distinctUntilChanged(
          (oldConfig, newConfig) =>
            oldConfig.langCode === newConfig.langCode &&
            oldConfig.darkMode === newConfig.darkMode &&
            oldConfig.analytics === newConfig.analytics
        ),
        map(config => ({
          locale: antdLocales[config.langCode] || en_US,
          darkMode: config.darkMode,
          analytics: config.analytics
        })),
        startWith({
          locale: zh_CN,
          darkMode: false,
          analytics: false
        })
      )
    )
  )!

  useEffect(() => {
    if (analytics && props.path) {
      reportGA(props.path)
    }
  }, [analytics, props.path])

  return (
    <I18nContextProvider>
      {darkMode && (
        <Helmet>{<link rel="stylesheet" href={darkTheme} />}</Helmet>
      )}
      <ReduxProvider store={storeRef.current}>
        <AntdConfigProvider locale={locale}>
          {props.children}
        </AntdConfigProvider>
        <SaladBowlContainer />
        <DictPanelContainer />
        <WordEditorContainer />
      </ReduxProvider>
    </I18nContextProvider>
  )
}
