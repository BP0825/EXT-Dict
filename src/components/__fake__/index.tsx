/*-----------------------------------------------*\
    Dicts
\*-----------------------------------------------*/

import setupEnv from './devDict'
import { getDefaultProfile, ProfileMutable } from '@/app-config/profiles'

const dict = 'cobuild'
const dictConfig = (getDefaultProfile() as ProfileMutable).dicts.all[dict]
dictConfig.options.cibaFirst = false

setupEnv({
  dict,
  dictConfig,
  text: 'how', // 当たる 吐く
})

/*-----------------------------------------------*\
    Wordpage
\*-----------------------------------------------*/

// import React from 'react'
// import ReactDOM from 'react-dom'
// import WordPage from '../WordPage'

// ReactDOM.render(<WordPage area='history' />, document.getElementById('root'))
