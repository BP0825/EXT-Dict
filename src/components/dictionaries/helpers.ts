import DOMPurify from 'dompurify'

export type HTMLString = string

export const enum SearchErrorType {
  NoResult,
  NetWorkError,
}

export function handleNoResult<T> (): Promise<T> {
  return Promise.reject(SearchErrorType.NoResult)
}

export function handleNetWorkError (): Promise<never> {
  return Promise.reject(SearchErrorType.NetWorkError)
}

export interface MachineTranslateResult {
  searchText: {
    text: string
    audio?: string
  }
  trans: {
    text: string
    audio?: string
  }
}

export function getText (parent: ParentNode, selector?: string): string {
  const child = selector ? parent.querySelector(selector) : parent
  if (!child) { return '' }
  return child['textContent'] || ''
}

export function getInnerHTMLThunk (
  host?: string,
  DOMPurifyConfig?: DOMPurify.Config,
): (parent: ParentNode, selector?: string) => HTMLString
export function getInnerHTMLThunk (...args) {
  let host: string = typeof args[0] === 'string' ? args.shift() : ''
  let DOMPurifyConfig: DOMPurify.Config = Object(args[0]) === args[0] ? args.shift() : {
    FORBID_TAGS: ['style'],
    FORBID_ATTR: ['style'],
  }

  if (host && !host.endsWith('/')) {
    host = host + '/'
  }
  return function getInnerHTML (parent: ParentNode, selector?: string): HTMLString {
    const child = selector ? parent.querySelector(selector) : parent
    if (!child) { return '' }
    let purifyResult = DOMPurify.sanitize(child['outerHTML'] || '', DOMPurifyConfig)
    const content = typeof purifyResult === 'string'
      ? purifyResult
      : purifyResult['outerHTML']
        ? purifyResult['outerHTML']
        : purifyResult.firstElementChild
          ? purifyResult.firstElementChild.outerHTML
          : ''
    return host
      ? content.replace(/href="\/[^/]/g, 'href="' + host)
      : content
  }
}

export function getOuterHTMLThunk (host?: string, DOMPurifyConfig?: DOMPurify.Config)
export function getOuterHTMLThunk (...args) {
  let host: string = typeof args[0] === 'string' ? args.shift() : ''
  let DOMPurifyConfig: DOMPurify.Config = Object(args[0]) === args[0] ? args.shift() : {
    FORBID_TAGS: ['style'],
    FORBID_ATTR: ['style'],
  }

  if (host && !host.endsWith('/')) {
    host = host + '/'
  }
  return function getOuterHTML (parent: ParentNode, selector?: string): HTMLString {
    const child = selector ? parent.querySelector(selector) : parent
    if (!child) { return '' }
    let purifyResult = DOMPurify.sanitize(child['innerHTML'] || '', DOMPurifyConfig)
    const content = typeof purifyResult === 'string'
      ? purifyResult
      : purifyResult['outerHTML']
        ? purifyResult['outerHTML']
        : purifyResult.firstElementChild
          ? purifyResult.firstElementChild.outerHTML
          : ''
    return host
      ? content.replace(/href="\/[^/]/g, 'href="' + host)
      : content
  }
}

export function decodeHEX (text: string): string {
  return text.replace(
    /\\x([0-9A-Fa-f]{2})/g,
    (m, p1) => String.fromCharCode(parseInt(p1, 16)),
  )
}

export function removeChild (parent: ParentNode, selector: string) {
  const child = parent.querySelector(selector)
  if (child) { child.remove() }
}

export function removeChildren (parent: ParentNode, selector: string) {
  parent.querySelectorAll(selector).forEach(el => el.remove())
}
