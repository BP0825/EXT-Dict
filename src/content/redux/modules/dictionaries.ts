export const enum SearchStatus {
  OnHold,
  Searching,
  Finished,
}

export type DictionariesState = {
}

const initState: DictionariesState = {
}

export default function reducer (state = initState, action) {
  return state
}
