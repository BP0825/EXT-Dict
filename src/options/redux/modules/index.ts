import { Action, ActionType, createReducer } from 'retux'
import {
  ThunkAction as CreateThunkAction,
  ThunkDispatch as CreateThunkDispatch
} from 'redux-thunk'
import { initState, State } from './state'
import { ActionCatalog } from './action-catalog'
import { actionHandlers } from './action-handlers'
import {
  useSelector as _useSelector,
  useDispatch as _useDispatch
} from 'react-redux'

export type StoreState = State

export type StoreActionCatalog = ActionCatalog

export type StoreActionType = ActionType<StoreActionCatalog>

export type StoreAction<T extends StoreActionType = StoreActionType> = Action<
  StoreActionCatalog,
  T
>

export type ThunkAction<
  Type extends StoreActionType = StoreActionType,
  Result = void
> = CreateThunkAction<
  Result,
  StoreState,
  never,
  Action<StoreActionCatalog, Type>
>

export type StoreDispatch<
  Type extends StoreActionType = StoreActionType
> = CreateThunkDispatch<StoreState, never, StoreAction<Type>>

export const useSelector: <TSelected = unknown>(
  selector: (state: StoreState) => TSelected,
  equalityFn?: (left: TSelected, right: TSelected) => boolean
) => TSelected = _useSelector

export const useDispatch: () => StoreDispatch = _useDispatch

export const getRootReducer = async () => {
  return createReducer(await initState(), actionHandlers)
}
