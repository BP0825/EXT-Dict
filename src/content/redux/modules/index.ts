import { combineReducers } from 'redux'
import config, { ConfigState } from './config'
import selection, { SelectionState } from './selection'
import dictionaries, { DictionariesState } from './dictionaries'
import widget, { WidgetState } from './widget'

export default combineReducers({
  config,
  selection,
  dictionaries,
  widget,
})

export type StoreState = {
  config: ConfigState
  selection: SelectionState
  dictionaries: DictionariesState
  widget: WidgetState
}
