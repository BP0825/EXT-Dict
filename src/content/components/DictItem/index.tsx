import React from 'react'
import { DictID } from '@/app-config'
import { translate } from 'react-i18next'
import { TranslationFunction } from 'i18next'
import { Spring } from 'react-spring'
import { openURL } from '@/_helpers/browser-api'

import { SearchStatus } from '@/content/redux/modules/dictionaries'

export interface DictItemDispatchers {
  readonly searchText: ({ id: DictID }) => any
  readonly updateItemHeight: ({ id, height }: { id: DictID, height: number }) => any
}

export interface DictItemProps extends DictItemDispatchers {
  readonly id: DictID
  readonly dictURL: string
  readonly fontSize: number
  readonly preferredHeight: number
  readonly panelWidth: number
  readonly searchStatus: SearchStatus
  readonly searchResult: any
  readonly isAnimation: boolean
}

export type DictItemState = {
  readonly propsSearchStatus: SearchStatus | null
  readonly propsSearchResult: any
  readonly offsetHeight: number
  readonly visibleHeight: number
  readonly isUnfold: boolean
}

export class DictItem extends React.PureComponent<DictItemProps & { t: TranslationFunction }, DictItemState> {
  bodyRef = React.createRef<HTMLDivElement>()
  prevItemHeight = 30
  initStyle = { height: 30, opacity: 0 }

  state = {
    /** same as pros */
    propsSearchStatus: null,
    propsSearchResult: null,
    /** real height of the body */
    offsetHeight: 10,
    /** final height, takes unfold and mask into account */
    visibleHeight: 10,
    isUnfold: false,
  }

  static getDerivedStateFromProps (
    nextProps: DictItemProps,
    prevState: DictItemState,
  ): Partial<DictItemState> | null {
    const {
      propsSearchResult: lastResult,
      propsSearchStatus: lastStatus,
    } = prevState
    const {
      searchResult: nextResult,
      searchStatus: nextStatus,
    } = nextProps

    if (lastResult === nextResult && lastStatus === nextStatus) {
      return null
    }

    switch (nextStatus) {
      case SearchStatus.OnHold:
      case SearchStatus.Searching:
        return {
          propsSearchStatus: nextStatus,
          propsSearchResult: nextResult,
          isUnfold: false,
          offsetHeight: 0,
          visibleHeight: 10,
        }
      case SearchStatus.Finished:
        return {
          propsSearchStatus: nextStatus,
          propsSearchResult: nextResult,
          isUnfold: true,
        }
    }

    return { propsSearchStatus: nextStatus, propsSearchResult: nextResult }
  }

  blurAfterClick (e) {
    e.currentTarget.blur()
  }

  calcBodyHeight (force?: boolean) {
    if (this.bodyRef.current) {
      const offsetHeight = Math.max(this.bodyRef.current.offsetHeight, 10) || 10
      if (force || this.state.offsetHeight !== offsetHeight) {
        return { offsetHeight, visibleHeight: Math.min(offsetHeight, this.props.preferredHeight) }
      }
    }
    return null
  }

  toggleFolding = () => {
    if (this.props.searchStatus === SearchStatus.Searching) {
      return
    }

    if (this.state.isUnfold) {
      this.setState({ isUnfold: false, visibleHeight: 10 })
    } else {
      if (this.props.searchResult) {
        const update = this.calcBodyHeight(true)
        if (update) {
          update['isUnfold'] = true
          this.setState(update)
        } else {
          this.setState({ isUnfold: true })
        }
      } else {
        this.props.searchText({ id: this.props.id })
      }
    }
  }

  showFull = e => {
    this.setState(state => ({ visibleHeight: state.offsetHeight }))
    e.currentTarget.blur()
  }

  openDictURL = e => {
    openURL(this.props.dictURL)
    e.preventDefault()
    e.stopPropagation()
  }

  componentDidUpdate (prevProps: DictItemProps) {
    if (this.props.searchStatus === SearchStatus.Finished &&
        this.props.searchStatus !== prevProps.searchStatus) {
      const update = this.calcBodyHeight()
      if (update) { this.setState(update as any) }
    }

    this.props.updateItemHeight({
      id: this.props.id,
      height: this.state.visibleHeight + 20,
    })
  }

  componentDidMount () {
    this.props.updateItemHeight({
      id: this.props.id,
      height: this.state.visibleHeight + 20,
    })
  }

  render () {
    const {
      t,
      id,
      dictURL,
      fontSize,
      searchStatus,
      searchResult,
      isAnimation,
    } = this.props

    const {
      offsetHeight,
      visibleHeight,
      isUnfold,
    } = this.state

    return (
      <section className='panel-DictItem'>
        <header className='panel-DictItem_Header' onClick={this.toggleFolding}>
          <img className='panel-DictItem_Logo' src={require('@/components/dictionaries/' + id + '/favicon.png')} alt='dict logo' />
          <h1 className='panel-DictItem_Title'>
            <a href={dictURL} onClick={this.openDictURL}>{t(`dict:${id}`)}</a>
          </h1>
          { searchStatus === SearchStatus.Searching &&
            <div className='panel-DictItem_Loader'>
              <div className='panel-DictItem_Loader_Ball' />
              <div className='panel-DictItem_Loader_Ball' />
              <div className='panel-DictItem_Loader_Ball' />
              <div className='panel-DictItem_Loader_Ball' />
              <div className='panel-DictItem_Loader_Ball' />
            </div>
          }
          <button className={'panel-DictItem_FoldArrowBtn'} onClick={this.blurAfterClick}>
            <svg className={`panel-DictItem_FoldArrow ${isUnfold ? 'isActive' : ''}`} width='18' height='18' viewBox='0 0 59.414 59.414' xmlns='http://www.w3.org/2000/svg'>
              <path d='M43.854 59.414L14.146 29.707 43.854 0l1.414 1.414-28.293 28.293L45.268 58' />
            </svg>
          </button>
        </header>
        <Spring from={this.initStyle} to={{ height: visibleHeight, opacity: isUnfold ? 1 : 0 }} immediate={!isAnimation}>
          {({ height, opacity }) => (
            <div className='panel-DictItem_Body'
              key={id}
              style={{ fontSize, height }}
            >
              <div ref={this.bodyRef} className='panel-DictItem_BodyMesure'>
                <article style={{ opacity }}>
                  {searchResult && React.createElement(require('@/components/dictionaries/' + id + '/View.tsx').default, { result: searchResult })}
                  {isUnfold &&
                    <button
                      className={`panel-DictItem_FoldMask ${visibleHeight < offsetHeight ? 'isActive' : ''}`}
                      onClick={this.showFull}
                    >
                      <svg className='panel-DictItem_FoldMaskArrow' width='15' height='15' viewBox='0 0 59.414 59.414' xmlns='http://www.w3.org/2000/svg'>
                        <path d='M58 14.146L29.707 42.44 1.414 14.145 0 15.56 29.707 45.27 59.414 15.56' />
                      </svg>
                    </button>
                  }
                </article>
              </div>
            </div>
          )}
        </Spring>
      </section>
    )
  }
}

export default translate()(DictItem)
