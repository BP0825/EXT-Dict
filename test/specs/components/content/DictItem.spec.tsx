import React from 'react'
import { mount, shallow } from 'enzyme'
import { SearchStatus } from '@/content/redux/modules/dictionaries'
import { DictItem, DictItemProps } from '@/content/components/DictItem'
import { TranslationFunction } from 'i18next'
import _ from 'lodash'

const mockDict = jest.fn(({ result }) => result ? 'rendered' : null)

jest.mock('@/components/dictionaries/bing/View.tsx', () => {
  return {
    default: mockDict
  }
})

type TransProps = DictItemProps & { t: TranslationFunction }

describe('Component/content/DictItem', () => {
  beforeEach(() => {
    mockDict.mockClear()
  })
  it('should render pending correctly', () => {
    const props: TransProps = {
      t: _.identity,
      id: 'bing',
      dictURL: 'https://google.com',
      fontSize: 14,
      searchResult: null,
      preferredHeight: 100,
      panelWidth: 400,
      searchStatus: SearchStatus.OnHold,
      updateItemHeight: _.noop,
      requestSearchText: _.noop
    }
    expect(mount(<DictItem {...props} />)).toMatchSnapshot()
  })

  it('should render result correctly', () => {
    const props: TransProps = {
      t: _.identity,
      id: 'bing',
      dictURL: 'https://google.com',
      fontSize: 14,
      searchResult: null,
      preferredHeight: 100,
      panelWidth: 400,
      searchStatus: SearchStatus.Searching,
      updateItemHeight: _.noop,
      requestSearchText: _.noop
    }
    const mounted = mount(<DictItem {...props} />)
    expect(mounted).toMatchSnapshot()
    expect(mounted.state('isUnfold')).toBe(false)

    mounted.setProps({ ...props, searchResult: 'result1', searchStatus: SearchStatus.Finished })
    expect(mounted).toMatchSnapshot()
    expect(mounted.state('isUnfold')).toBe(true)
    setTimeout(() => {
      expect(mockDict).toHaveBeenCalledWith(expect.objectContaining({ result: 'result1' }))
    }, 0)
  })

  it('toggling unfold', () => {
    const props: TransProps = {
      t: _.identity,
      id: 'bing',
      dictURL: 'https://google.com',
      fontSize: 14,
      searchResult: null,
      preferredHeight: 100,
      panelWidth: 400,
      searchStatus: SearchStatus.Searching,
      updateItemHeight: _.noop,
      requestSearchText: jest.fn()
    }
    const shallowed = shallow(<DictItem {...props} />)
    expect(shallowed.state('isUnfold')).toBe(false)

    shallowed.setProps({ ...props, searchStatus: SearchStatus.Finished })
    expect(shallowed.state('isUnfold')).toBe(true)

    shallowed.find('header').simulate('click')
    expect(shallowed.state('isUnfold')).toBe(false)

    // request searching
    expect(props.requestSearchText).toHaveBeenCalledTimes(0)
    shallowed.find('header').simulate('click')
    expect(shallowed.state('isUnfold')).toBe(false)
    expect(props.requestSearchText).toHaveBeenCalledTimes(1)
  })
})
