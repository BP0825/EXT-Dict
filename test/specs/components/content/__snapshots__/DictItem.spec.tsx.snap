// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/content/DictItem should render pending correctly 1`] = `
<DictItem
  dictURL="https://google.com"
  fontSize={14}
  id="bing"
  panelWidth={400}
  preferredHeight={100}
  requestSearchText={[Function]}
  searchResult={null}
  searchStatus={0}
  t={[Function]}
>
  <section
    className="panel-DictItem"
  >
    <header
      className="panel-DictItem_Header"
      onClick={[Function]}
    >
      <img
        alt="dict logo"
        className="panel-DictItem_Logo"
        src="favicon.png"
      />
      <h1
        className="panel-DictItem_Title"
      >
        <a
          className="panel-DictItem_Title"
          href="https://google.com"
          onClick={[Function]}
        >
          dict_bing
        </a>
      </h1>
      <button
        className="panel-DictItem_FoldArrowBtn"
        onClick={[Function]}
      >
        <svg
          className="panel-DictItem_FoldArrow "
          height="18"
          viewBox="0 0 59.414 59.414"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M43.854 59.414L14.146 29.707 43.854 0l1.414 1.414-28.293 28.293L45.268 58"
          />
        </svg>
      </button>
    </header>
    <Motion
      defaultStyle={
        Object {
          "height": 0,
          "opacity": 0,
        }
      }
      style={
        Object {
          "height": Object {
            "damping": 26,
            "precision": 0.01,
            "stiffness": 170,
            "val": 0,
          },
          "opacity": Object {
            "damping": 26,
            "precision": 0.01,
            "stiffness": 170,
            "val": 0,
          },
        }
      }
    >
      <div
        className="panel-DictItem_Body"
        style={
          Object {
            "fontSize": 14,
            "height": 0,
          }
        }
      >
        <article
          style={
            Object {
              "opacity": 0,
            }
          }
        >
          <mockConstructor
            result={null}
          />
          <button
            className="panel-DictItem_FoldMask "
            onClick={[Function]}
          >
            <svg
              className="panel-DictItem_FoldMaskArrow"
              height="15"
              viewBox="0 0 59.414 59.414"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M58 14.146L29.707 42.44 1.414 14.145 0 15.56 29.707 45.27 59.414 15.56"
              />
            </svg>
          </button>
        </article>
      </div>
    </Motion>
  </section>
</DictItem>
`;

exports[`Component/content/DictItem should render result correctly 1`] = `
<DictItem
  dictURL="https://google.com"
  fontSize={14}
  id="bing"
  panelWidth={400}
  preferredHeight={100}
  requestSearchText={[Function]}
  searchResult={null}
  searchStatus={1}
  t={[Function]}
>
  <section
    className="panel-DictItem"
  >
    <header
      className="panel-DictItem_Header"
      onClick={[Function]}
    >
      <img
        alt="dict logo"
        className="panel-DictItem_Logo"
        src="favicon.png"
      />
      <h1
        className="panel-DictItem_Title"
      >
        <a
          className="panel-DictItem_Title"
          href="https://google.com"
          onClick={[Function]}
        >
          dict_bing
        </a>
      </h1>
      <div
        className="panel-DictItem_Loader"
      >
        <div
          className="panel-DictItem_Loader_Ball"
        />
        <div
          className="panel-DictItem_Loader_Ball"
        />
        <div
          className="panel-DictItem_Loader_Ball"
        />
        <div
          className="panel-DictItem_Loader_Ball"
        />
        <div
          className="panel-DictItem_Loader_Ball"
        />
      </div>
      <button
        className="panel-DictItem_FoldArrowBtn"
        onClick={[Function]}
      >
        <svg
          className="panel-DictItem_FoldArrow "
          height="18"
          viewBox="0 0 59.414 59.414"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M43.854 59.414L14.146 29.707 43.854 0l1.414 1.414-28.293 28.293L45.268 58"
          />
        </svg>
      </button>
    </header>
    <Motion
      defaultStyle={
        Object {
          "height": 0,
          "opacity": 0,
        }
      }
      style={
        Object {
          "height": Object {
            "damping": 26,
            "precision": 0.01,
            "stiffness": 170,
            "val": 0,
          },
          "opacity": Object {
            "damping": 26,
            "precision": 0.01,
            "stiffness": 170,
            "val": 0,
          },
        }
      }
    >
      <div
        className="panel-DictItem_Body"
        style={
          Object {
            "fontSize": 14,
            "height": 0,
          }
        }
      >
        <article
          style={
            Object {
              "opacity": 0,
            }
          }
        >
          <mockConstructor
            result={null}
          />
          <button
            className="panel-DictItem_FoldMask "
            onClick={[Function]}
          >
            <svg
              className="panel-DictItem_FoldMaskArrow"
              height="15"
              viewBox="0 0 59.414 59.414"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M58 14.146L29.707 42.44 1.414 14.145 0 15.56 29.707 45.27 59.414 15.56"
              />
            </svg>
          </button>
        </article>
      </div>
    </Motion>
  </section>
</DictItem>
`;

exports[`Component/content/DictItem should render result correctly 2`] = `
<DictItem
  dictURL="https://google.com"
  fontSize={14}
  id="bing"
  panelWidth={400}
  preferredHeight={100}
  requestSearchText={[Function]}
  searchResult="result1"
  searchStatus={2}
  t={[Function]}
>
  <section
    className="panel-DictItem"
  >
    <header
      className="panel-DictItem_Header"
      onClick={[Function]}
    >
      <img
        alt="dict logo"
        className="panel-DictItem_Logo"
        src="favicon.png"
      />
      <h1
        className="panel-DictItem_Title"
      >
        <a
          className="panel-DictItem_Title"
          href="https://google.com"
          onClick={[Function]}
        >
          dict_bing
        </a>
      </h1>
      <button
        className="panel-DictItem_FoldArrowBtn"
        onClick={[Function]}
      >
        <svg
          className="panel-DictItem_FoldArrow isActive"
          height="18"
          viewBox="0 0 59.414 59.414"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M43.854 59.414L14.146 29.707 43.854 0l1.414 1.414-28.293 28.293L45.268 58"
          />
        </svg>
      </button>
    </header>
    <Motion
      defaultStyle={
        Object {
          "height": 0,
          "opacity": 0,
        }
      }
      style={
        Object {
          "height": Object {
            "damping": 26,
            "precision": 0.01,
            "stiffness": 170,
            "val": 0,
          },
          "opacity": Object {
            "damping": 26,
            "precision": 0.01,
            "stiffness": 170,
            "val": 1,
          },
        }
      }
    >
      <div
        className="panel-DictItem_Body"
        style={
          Object {
            "fontSize": 14,
            "height": 0,
          }
        }
      >
        <article
          style={
            Object {
              "opacity": 0,
            }
          }
        >
          <mockConstructor
            result="result1"
          >
            rendered
          </mockConstructor>
          <button
            className="panel-DictItem_FoldMask "
            onClick={[Function]}
          >
            <svg
              className="panel-DictItem_FoldMaskArrow"
              height="15"
              viewBox="0 0 59.414 59.414"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M58 14.146L29.707 42.44 1.414 14.145 0 15.56 29.707 45.27 59.414 15.56"
              />
            </svg>
          </button>
        </article>
      </div>
    </Motion>
  </section>
</DictItem>
`;
